src/kronecker.jl

```diff
@@ -181,20 +193,20 @@ end
 ###############
 # KroneckerSumMap
 ###############
-struct KroneckerSumMap{T, As<:Tuple{LinearMap,LinearMap}} <: LinearMap{T}
+struct KroneckerSumMap{T, As<:Tuple{LinearMap, LinearMap}} <: LinearMap{T}
     maps::As
-    function KroneckerSumMap{T, As}(maps::As) where {T, As}
-        for n in eachindex(maps)
-            A = maps[n]
-            size(A, 1) == size(A, 2) || throw(ArgumentError("operators need to be square in Kronecker sums"))
```

Comment:
🤦

Reply:
Is there a reason actually that `KroneckerSum` is restricted to two arguments. If it is like the logarithm of a Kronecker product, it could also have an arbitrary number of factors, or rather, terms, right? I didn't know this was a thing even until you implemented it, although I do encounter it also in my research field.

Reply:
This question made me curious, so I have just verified that it is associative.  (It is not commutative.)
So it can extend to more arguments, though I think the easiest construction is probably recursive.
Not to much about this that I could find online other than this:
https://www.euclideanspace.com/maths/algebra/multidimensional/kroneckerSum/index.htm 
And that is an unreliable source because it wrongly claims commutativity...

Reply:
Surely, I must have been considering this, given the KroneckerMap design above. You can `kronsum` several maps, but for implementation reasons that I don't remember, for instance, `kronsum(A, B, C) = kronsum(kronsum(A, B), C)` simply means you have an "inner" and an "outer" KroneckerSumMap. Given the nearly perfect coverage, one could try to rewrite things gently internally to flatten that structure, but this doesn't have high priority, I'd say.